modules/transaction-payment/src/lib.rs

```diff
@@ -567,17 +567,24 @@ where
 		charge_fee_order.dedup();
 
 		let price_impact_limit = Some(T::MaxSlippageSwapWithDEX::get());
+		let native_existential_deposit = <T as Config>::Currency::minimum_balance();
+		let total_native = <T as Config>::Currency::total_balance(who);
+		// add the gap amount to keep account alive and have enough fee
+		let fee_and_alive_gap = if total_native < native_existential_deposit {
+			fee.saturating_add(native_existential_deposit.saturating_sub(total_native))
+		} else {
+			fee
+		};
 
 		// iterator charge fee order to get enough fee
 		for currency_id in charge_fee_order {
 			if currency_id == native_currency_id {
 				// check native balance if is enough
-				let native_is_enough =
-					<T as Config>::Currency::free_balance(who)
-						.checked_sub(&fee)
-						.map_or(false, |new_free_balance| {
-							<T as Config>::Currency::ensure_can_withdraw(who, fee, reason, new_free_balance).is_ok()
-						});
+				let native_is_enough = <T as Config>::Currency::free_balance(who)
+					.checked_sub(&fee_and_alive_gap)
+					.map_or(false, |new_free_balance| {
+						<T as Config>::Currency::ensure_can_withdraw(who, fee, reason, new_free_balance).is_ok()
```

Comment:
```suggestion
						<T as Config>::Currency::ensure_can_withdraw(who, fee, reason, fee_and_alive_gap).is_ok()
```

Reply:
@xlc The old code seems to be wrong.

Reply:
https://github.com/paritytech/substrate/blob/993907f9deb180795b291ba752b9c204165d74dc/frame/support/src/traits/tokens/currency.rs#L103

the last parameter is expected new balance after withdraw

Reply:
Ok, It's different from tokens.
https://github.com/open-web3-stack/open-runtime-module-library/blob/3d64077e5ac5539bde336a794308d85f6f5f2b82/tokens/src/lib.rs#L642

Reply:
Yes we removed the last parameter